aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1624 @@
+        owner: &signer, multisig_account: address, sequence_number: u64) acquires MultisigAccount {
+        vote_transanction(owner, multisig_account, sequence_number, true);
+    }
+
+    /// Reject a multisig transaction.
+    public entry fun reject_transaction(
+        owner: &signer, multisig_account: address, sequence_number: u64) acquires MultisigAccount {
+        vote_transanction(owner, multisig_account, sequence_number, false);
+    }
+
+    /// Generic function that can be used to either approve or reject a multisig transaction
+    public entry fun vote_transanction(
+        owner: &signer, multisig_account: address, sequence_number: u64, approved: bool) acquires MultisigAccount {
+        assert_multisig_account_exists(multisig_account);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        assert_is_owner(owner, multisig_account_resource);
+
+        assert!(
+            table::contains(&multisig_account_resource.transactions, sequence_number),
+            error::not_found(ETRANSACTION_NOT_FOUND),
+        );
+        let transaction = table::borrow_mut(&mut multisig_account_resource.transactions, sequence_number);
+        let votes = &mut transaction.votes;
+        let owner_addr = address_of(owner);
+
+        if (simple_map::contains_key(votes, &owner_addr)) {
+            *simple_map::borrow_mut(votes, &owner_addr) = approved;
+        } else {
+            simple_map::add(votes, owner_addr, approved);
+        };
+
+        emit_event(
+            &mut multisig_account_resource.vote_events,
+            VoteEvent {
+                owner: owner_addr,
+                sequence_number,
+                approved,
+            }
+        );
+    }
+
+    /// Remove the next transaction if it has sufficient owner rejections.
+    public entry fun execute_rejected_transaction(
+        owner: &signer,
+        multisig_account: address,
+    ) acquires MultisigAccount {
+        assert_multisig_account_exists(multisig_account);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        assert_is_owner(owner, multisig_account_resource);
+        let sequence_number = multisig_account_resource.last_executed_sequence_number + 1;
+        assert!(
+            table::contains(&multisig_account_resource.transactions, sequence_number),
+            error::not_found(ETRANSACTION_NOT_FOUND),
+        );
+        // Delete the transaction to keep storage efficient. Off-chain components can reconstruct the transaction and
+        // from events.
+        let transaction = table::remove(&mut multisig_account_resource.transactions, sequence_number);
+        let (_, num_rejections) = num_approvals_and_rejections(&multisig_account_resource.owners, &transaction);
+        assert!(
+            num_rejections >= multisig_account_resource.num_signatures_required,
+            error::invalid_state(ENOT_ENOUGH_REJECTIONS),
+        );
```

Comment:
this is the same as remove_executed_transaction? we just need to return (num_approvals, num_rejections)

Reply:
That's true. Should be an easy refactor. I can do this in a followup PR